### PR TITLE
Update NFS client helm chart

### LINKED ISSUE #480: Confused about `nfs-provisioner` app
Somewhere along the way, I was lead to believe that the NFS provisioner app is `nfs-provisioner`, but is actually `nfs-client-provisioner`. 

https://github.com/alexellis/arkade/blob/3be9b511ca1625f2b07b26df49e85bcf760b9fad/cmd/install.go#L95

**Example:**  _running this command produces no output_
```shell
❯ ark i nfs-provisioner
```

The list indicates it is available as `nfs-client-provisioner`. 
```shell
❯ ark install --help

Available Commands:
  ...
  nfs-client-provisioner  Install nfs client provisioner
```

### PR DESCRIPTION
**Pull request by Shikachuu**
Add alias for the original name
Rename the chart
Fix helm url
Update info msg

Signed-off-by: Czékus Máté <mate@picloud.hu>

## Description
The `nfs-client-provisioner` chart was so outdated it has been moved twice and renamed since the `https://charts.helm.sh/stable` deprecation.
I also added the original name (`nfs-client-provisioner`) to maintain backward compatibility.

## Motivation and Context
- [X] I have raised an issue to propose this change, which has been given a label of `design/approved` by a maintainer ([required](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md)) #274
Closes #480 with the new alias.

## How Has This Been Tested?
By running `ark install nfs-subdir-external-provisioner` and `ark install nfs-client-provisioner`, after that check the helm releases.
image

I know it sounds weird, but the chart itself functionally has not changed, nor the values.
## Are you a GitHub Sponsor yet (Yes/No?)

- [ ] Yes
- [X] No

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Documentation

- [ ] I have updated the list of tools in README.md if (required) with `./arkade get -o markdown`
- [X] I have updated the list of apps in README.md if (required) with `./arkade install --help`

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I've read the [CONTRIBUTION](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md) guide
- [X] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

- [X] I have tested this on arm, or have added code to prevent deployment

Also in the changes there is a removed arm check, since the [chart has arm v7 and arm64 binaries.](https://github.com/kubernetes-sigs/nfs-subdir-external-provisioner#build-and-publish-with-github-actions=)
I also tested this on an arm64 cluster.

### CODE REVIEW

File: cmd/apps/nfs_app.go [hunk 1]
```diff
@@ -9,16 +9,16 @@ import (
...
 	"github.com/alexellis/arkade/pkg/types"
 	"github.com/spf13/cobra"
 )
 
 func MakeInstallNfsProvisioner() *cobra.Command {
 	var nfsProvisionerApp = &cobra.Command{
-		Use:          "nfs-client-provisioner",
-		Short:        "Install nfs client provisioner",
-		Long:         "Install nfs client provisioner to create dynamic persistent volumes",
+		Use:          "nfs-subdir-external-provisioner",
```
Comment (alexellis) [inline cmd/apps/nfs_app.go:18]: What a long time. Glad you have the alias.

Perhaps one more alias of `nfs-provisioner` could work?

**Review by alexellis — APPROVED** [review]
Approved

File: cmd/apps/nfs_app.go (hunk 1 above)
Comment (Shikachuu) [reply cmd/apps/nfs_app.go:18]: Sure, good idea, I've added it.

**Review by alexellis — APPROVED** [review]
Approved

**Pull request merged**